command/command_test.go

```diff
@@ -0,0 +1,77 @@
+package command
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel"
+)
+
+func TestExecuteProgram(t *testing.T) {
+
+	runfilePath := "command/tests/tests_/tests"
+	testSampleCodePath, err := bazel.Runfile(runfilePath)
+	if err != nil {
+		t.Errorf("Encountered error %s by bazel.Runfile with arg %s", err.Error(), runfilePath)
+	}
+	test := ExternalCommand{
+		Path: testSampleCodePath,
+		Env:  os.Environ(),
+	}
+
+	programstate, err := ExecuteProgram(test, testErrorHandler)
+	if err != nil {
+		t.Errorf("Encountered error %s by ExecuteProgram()", err.Error())
+	}
+
+	want := "Hello\n"
+	got := testWaitOutput(programstate.Stdout, want)
+
+	serr := programstate.Stderr()
+	if len(serr) != 0 {
+		fmt.Println("Returned error by StderrNonBlocking()", string(serr))
+	}
+
+	if got != want {
+		t.Errorf("Got text = %q, want %q", got, want)
+	}
+
+	err = programstate.Command.Process.Signal(os.Interrupt)
+	if err != nil {
+		t.Errorf("Encountered error %s while sending interrupt signal", err.Error())
+	}
+
+	want = "Hello\nWorld\n"
+	got = testWaitOutput(programstate.Stdout, want)
```

Comment:
This works, but a cleaner way would be to modify the function to return err == nil if the output does match within timeout and a new error with the error message if it doesn't. then move this function & combine it with if statement.
Right now both your loop, and the got != want below are checking for content equality.